Writerside/topics/raptor.md

```diff
@@ -1,49 +1,61 @@
-visit in the current lap, or when a certain number of laps have been completed.
-
-**Tracing the route**: After the algorithm has finished, the optimal route can be found by
-backtracking the updated arrival
-times from the destination station to the start station.
-
-The **RAPTOR** algorithm is very efficient because it takes advantage of the fact that public
-transportation systems usually have regular schedules and that it is often faster to stay on
-the same line instead of switching. It is particularly well suited for real-time
-routing and planning in large transportation networks.
-
-## Flowchart
+The **RAPTOR** (Round-based Public Transit Routing) algorithm is a powerful algorithm to compute optimal routes in
+public transportation systems, especially for large and complex networks. Its primary objective is to efficiently
+determine the best route options based on factors like arrival time and number of transfers. RAPTOR is particularly
+useful for real-time journey planning due to its speed and adaptability.
+
+### Key Concepts of RAPTOR:
+
+- **Initialization**: The algorithm starts at the origin station and time specified by the user. Initially, it only
+  considers the starting station, marking it for the first round of exploration.
+
+- **Round-based Search**: The search process is broken into several rounds. In each round, the algorithm evaluates all
+  transport lines connected to the current list of stations. It computes the earliest possible arrival times at the
+  stations along each line and adds them to the list of stations for the next round.
+
+- **Updating Arrival Times**: As the algorithm processes new stations, it updates their earliest arrival times. If it
+  finds a route to a station that is faster than the previously known route, the arrival time and corresponding route
+  are updated.
+
+- **Termination Condition**: The algorithm stops when it reaches the destination station with the optimal arrival time
+  or when no further improvements can be made after a set number of rounds.
+
+- **Tracing the Optimal Route**: Once the algorithm concludes, the optimal route is reconstructed by tracing the updated
+  arrival times back from the destination to the start station.
+
+RAPTOR is highly efficient because it leverages the structure of public transit systems, where it is often faster to
+remain on the same line rather than switching between multiple lines. This characteristic makes RAPTOR particularly
+suitable for large-scale transportation networks where real-time routing is essential.
+
+## Sequence Diagram RAPTOR
 
 ```mermaid
-graph TD
-    A[Start at departure stop] --> B{Round-based search}
-    B -->|Found new stop| C[Update earliest arrival time]
-    C --> D{Check for termination}
-    D -->|Not terminated| B
-    D -->|Terminated| E[Backtrack route]
-    E --> F[End]
+sequenceDiagram
+    participant User
+    participant Algorithm
+    participant Stops
+    participant Routes
+    participant Timetable
```

Comment:
I would rename Timetable to Schedule, because we call it schedule in the code base and I think it's also more precise naming.

Reply:
Actually what the sequence diagram referers to is `stop times` or if you want to simplify `trips`.